### PR TITLE
Add password strength meter

### LINKED ISSUE #233: Add password strength component
* Create a linode
* Rebuild
* Rescue (resize a password)

### PR DESCRIPTION
**Pull request by SirCmpwn**
Closes #233 

Still needs tests.

### CODE REVIEW

**Comment by eatonphil** [comment]
Very nice.

**Comment by SirCmpwn** [comment]
Actually good point @coveralls. Not sure if more testing is necessary here, all this code is already hit from other tests. It's mostly styles. Anyone else have an opinion on that?

File: src/linodes/create/components/Details.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ export default class Details extends Component {
             onChange={e => this.setState({ enableBackups: e.target.value })}
             name="enableBackups"
           />
-          Enable ($2.50/month)
+          <span>Enable ($2.50/month)</span>
```
Comment (eatonphil) [inline src/linodes/create/components/Details.js]: +1

File: src/linodes/create/components/Details.js [hunk 2]
```diff
@@ -56,20 +69,35 @@ export default class Details extends Component {
...
+          autoComplete="off"
         />
         <button
           type="button"
           className="btn btn-secondary"
           onClick={() =>
-            this.setState({ password: generatePassword(30, false) })}
+            this.onPasswordChange({ target: { value: generatePassword(30, false) } })}
         >Generate</button>
+        <div className={`strength strength-${this.state.strength.score}`}>
```
Comment (eatonphil) [inline src/linodes/create/components/Details.js]: Can we pull this into a component? This will definitely be used elsewhere.

Comment (SirCmpwn) [inline src/linodes/create/components/Details.js]: I'd rather do that when we come to the point where we need to reuse it. I'm not sure of any other products we currently offer that would need this.

Comment (eatonphil) [inline src/linodes/create/components/Details.js]: That's fine I guess. From #233 though there is also rebuild and rescue.

Comment (SirCmpwn) [inline src/linodes/create/components/Details.js]: Ah, fair enough. Let's just estimate that into the scope of #233 though, there are some complications here that make me hesitate to pull it out.

File: scss/linodes/create/create.scss [hunk 3]
```diff
@@ -8,3 +8,50 @@
...
+
+        &.strength-2 {
+            span:nth-child(1), span:nth-child(2) {
+                background: $red;
+            }
+        }
+
+        &.strength-3 {
+            span:nth-child(1), span:nth-child(2), span:nth-child(3) {
+                background: darken($yellow, 20);
```
Comment (eatonphil) [inline scss/linodes/create/create.scss]: Not a big fan of making arbitrary changes to variables at random places. This promotes inconsistency everywhere we feel like the current colors don't work.  So you could darken yellow here by 20 but the next time you may forget and darken it by 25 or 15. So instead just create a dark-yellow color at the top-level and just use that?

Comment (SirCmpwn) [inline scss/linodes/create/create.scss]: Alright

**Comment by eatonphil** [comment]
ty

**Comment by SirCmpwn** [comment]
Ready to rock

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**